app/helpers.go

```diff
@@ -112,10 +113,11 @@ func (s *SliceIterator) Valid() bool {
 // Next moves the iterator to the next sequential key in the database, as
 // defined by order of iteration.
 //
-// If Valid returns false, this method will panic.
-func (s *SliceIterator) Next() {
+// If Valid returns false, this method will panic.keys
+func (s *SliceIterator) Next() error {
 	s.assertValid()
 	s.idx++
+	return nil
```

Comment:
`assertValid` is calling `panic`. It could return an error instead as well.

Reply:
:+1: